src/GraphQL/Server.hs

```diff
@@ -274,22 +312,27 @@ instance forall typeName interfaces fields m.
 -- | Closed type family to enforce the invariant that Union types
 -- contain only Objects.
 type family RunUnionType m (a :: [Type]) :: Type where
-  RunUnionType m (Object typeName interfaces fields:rest) = Handler m (Object typeName interfaces fields) :<|> RunUnionType m rest
-  RunUnionType m '[] = ()
+  RunUnionType m '[Object typeName interfaces fields] = Object typeName interfaces fields
+  RunUnionType m (Object typeName interfaces fields:rest) = Object typeName interfaces fields :<|> RunUnionType m rest
```

Comment:
Let's stick without parens for now. That's what `hlint` does, and it's a de facto standard.